.github/workflows/storybook.yml

```diff
@@ -0,0 +1,86 @@
+name: Storybook
+
+env:
+  AWS_S3_BUCKET: ${{ secrets.AWS_S3_BUCKET_MYCRYPTOBUILDS }}
+  AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID_MYCRYPTOBUILDS }}
+  AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY_MYCRYPTOBUILDS }}
+  AWS_REGION: 'us-east-2'
+
+on:
+  pull_request:
+    branches:
+      - '*'
+  push:
+    branches:
+      - master
+
+jobs:
+  storybook-build:
+    name: Storybook Build
+    runs-on: ubuntu-latest
+
+    steps:
+      # Fetch the PR number to use it with AWS
+      - name: Set PR_NUMBER
+        run: |
+          PR_NUMBER=$(echo $GITHUB_REF | awk 'BEGIN { FS = "/" } ; { print $3 }')
+          echo "PR_NUMBER=$PR_NUMBER" >> $GITHUB_ENV
+
+      - uses: actions/checkout@v2
+
+      - name: Read .nvmrc
+        run: echo "##[set-output name=NVMRC;]$(cat .nvmrc)"
+        id: nvm
+
+      - name: Setup Node.js ${{ steps.nvm.outputs.NVMRC }}
+        uses: actions/setup-node@v2.1.5
+        with:
+          node-version: ${{ steps.nvm.outputs.NVMRC }}
+
+      - name: Cache Dependencies
+        uses: actions/cache@v2.1.4
+        with:
+          path: node_modules
+          key: yarn-${{ hashFiles('yarn.lock') }}
+
+      - name: Install Dependencies
+        run: yarn install --frozen-lockfile --ignore-scripts
+
+      - name: Build Storybook
+        run: yarn build:storybook
+
+      - name: S3 Sync - Storybook SHA
+        uses: jakejarvis/s3-sync-action@v0.5.1
+        if: github.event_name == 'pull_request'
+        with:
+          args: --acl public-read --cache-control max-age=0 --follow-symlinks --exclude '.git/*'
+        env:
+          DEST_DIR: 'storybook/${GITHUB_SHA}'
```

Comment:
I would recommend changing all `DEST_DIR` to include a ui prefix -

`'ui/storybook/${GITHUB_SHA}'`